src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs

```diff
@@ -170,11 +170,20 @@ public int MaxModelValidationErrors
         public bool AllowBindingUndefinedValueToEnumType { get; set; }
 
         /// <summary>
-        /// Gets or sets the option to determine if model binding should convert all exceptions(including ones not related to bad input)
+        /// Gets or sets the option to determine if model binding should convert all exceptions (including ones not related to bad input)
         /// that occur during deserialization in <see cref="IInputFormatter"/>s into model state errors.
         /// This option applies only to custom <see cref="IInputFormatter"/>s.
         /// Default is <see cref="InputFormatterExceptionModelStatePolicy.AllExceptions"/>.
         /// </summary>
         public InputFormatterExceptionModelStatePolicy InputFormatterExceptionModelStatePolicy { get; set; }
+
+        /// <summary>
+        /// Gets or sets a flag to determine whether, if an action receives invalid JSON in
+        /// the request body, the JSON deserialization exception message should be added
+        /// to model state.
+        /// <see langword="true"/> by default, meaning that clients may receive details about
+        /// why the JSON they posted is considered invalid.
+        /// </summary>
+        public bool UseJsonDeserializationExceptionMessagesInModelState { get; set; } = true;
```

Comment:
Would it be possible to state this as a negative? IE `SuppressJsonDeserialization...` and default to false?

The world is a simpler place when flags default to false.

Reply:
OK, done